src/api/index.ts

```diff
@@ -3,96 +3,82 @@ import {
 );
 
+const request = async (path: string, init?: RequestInit) => {
+  const base =
+    SERVERS[
+      getLocalStorage(KEY_LOCALSTORAGE_SERVER_OWNER, DEFAULT_VALUE_SERVER_OWNER)
+    ];
+  const response = await fetch([base, path].join(""), {
+    ...init,
+    headers: {
+      Authorization: `Basic ${base64}`,
+      "Content-Type": "application/json",
+      ...init?.headers,
+    },
+  });
+
+  if (!response.ok) throw new Error(response.status.toString());
+  return response;
+};
+
 export const fetchProducts = () =>
-  fetch(
-    `${
-      SERVERS[
-        getLocalStorage(
-          KEY_LOCALSTORAGE_SERVER_OWNER,
-          DEFAULT_VALUE_SERVER_OWNER
-        )
-      ]
-    }/products`
-  );
+  request("/products").then((response) => response.json());
 
-export const fetchCartItems = async () =>
-  fetch(
-    `${
-      SERVERS[
-        getLocalStorage(
-          KEY_LOCALSTORAGE_SERVER_OWNER,
-          DEFAULT_VALUE_SERVER_OWNER
-        )
-      ]
-    }/cart-items`,
-    {
-      headers: {
-        Authorization: `Basic ${base64}`,
-      },
-    }
-  );
+export const fetchCartItems = () =>
+  request(`/cart-items`).then((response) => response.json());
 
 export const changeQuantity = async (cartItemId: number, newQuantity: number) =>
-  fetch(
-    `${
-      SERVERS[
-        getLocalStorage(
-          KEY_LOCALSTORAGE_SERVER_OWNER,
-          DEFAULT_VALUE_SERVER_OWNER
-        )
-      ]
-    }/cart-items/${cartItemId}`,
-    {
-      headers: {
-        Authorization: `Basic ${base64}`,
-        "Content-Type": "application/json",
-      },
-      method: "PATCH",
-      body: JSON.stringify({ quantity: newQuantity }),
-    }
-  );
+  request(`/cart-items/${cartItemId}`, {
+    method: "PATCH",
+    body: JSON.stringify({ quantity: newQuantity }),
+  });
 
 export const addCartItem = async (productId: number) =>
-  fetch(
-    `${
-      SERVERS[
-        getLocalStorage(
-          KEY_LOCALSTORAGE_SERVER_OWNER,
-          DEFAULT_VALUE_SERVER_OWNER
-        )
-      ]
-    }/cart-items`,
-    {
-      headers: {
-        Authorization: `Basic ${base64}`,
-        "Content-Type": "application/json",
-      },
-      method: "POST",
-      body: JSON.stringify({ productId: productId }),
-    }
-  );
+  request(`/cart-items`, {
+    method: "POST",
+    body: JSON.stringify({ productId: productId }),
+  });
 
 export const deleteCartItem = async (cartItemId: number) =>
-  fetch(
-    `${
-      SERVERS[
-        getLocalStorage(
-          KEY_LOCALSTORAGE_SERVER_OWNER,
-          DEFAULT_VALUE_SERVER_OWNER
-        )
-      ]
-    }/cart-items/${cartItemId}`,
-    {
-      headers: {
-        Authorization: `Basic ${base64}`,
-      },
-      method: "DELETE",
-    }
+  request(`/cart-items/${cartItemId}`, {
+    method: "DELETE",
+  });
+
+export const fetchCoupons = async (cartItemIdList: number[]) => {
+  const orderQuery = cartItemIdList
+    .map((el) => {
+      return `cartItemId=${el}`;
+    })
```

Comment:
소소하지만 `el`이라고 명명하신 이유가 있나요?

아래처럼 작성하면 더 직관적일거 같네요

```suggestion
    .map((id) => {
      return `cartItemId=${id}`;
    })
```

Reply:
아 훨씬 더 직관적이네요! 수정했습니다!